foursight_core/checks/access_key_expiration_detection.py

```diff
@@ -69,10 +69,18 @@ def refresh_access_keys(connection, **kwargs):
                                       f'&sort=-date_created', key=connection.ff_keys)
         # generate new key
         access_key_req = {'user': user_uuid, 'description': kp_name}
-        access_key_res = post_metadata(access_key_req, 'access-keys', key=connection.ff_keys)['@graph'][0]
-        s3_obj = {'secret': access_key_res['secret_access_key'],
-                  'key': access_key_res['access_key_id'],
-                  'server': s3.url}
+        # 2020-06-13/dmichaels: The actual result returned by the portal for this POST is not what
+        # seems to be expected; the access_key_id and secret_access_key are not within the @graph
+        # array; but handle both cases just in case; maybe that as an older (or newer) API.
+        # access_key_res = post_metadata(access_key_req, 'access-keys', key=connection.ff_keys)['@graph'][0]
+        access_key_res = post_metadata(access_key_req, 'access-keys', key=connection.ff_keys)
+        access_key_id = access_key_res['access_key_id'] if 'access_key_id' in access_key_res else None
+        secret_access_key = access_key_res['secret_access_key'] if 'secret_access_key' in access_key_res else None
+        if not access_key_id:
+            access_key_id = access_key_res['@graph'][0]['access_key_id']
+        if not secret_access_key:
+            secret_access_key = access_key_res['@graph'][0]['secret_access_key']
```

Comment:
Untested, and not precisely semantically equivalent, but perhaps a simplification if the problem is that this either returns a search result or a value.
```suggestion
        if '@graph' in access_key_res:  # Accommodate lack of clarity in return format
            access_key_res = access_key_res['@graph'][0]
        access_key_id = access_key_res.get('access_key_id')
        secret_access_key = access_key_res.get('secret_access_key')
```
I thought the only things that return @graph results was searches and collections, which can have multiple results, but I don't do a lot of these things. I thought get and post don't return graphs. @willronchetti do you have an opinion here?

Reply:
Yeah I'm not sure. And I should point out that I guess it's *possible* that I'm hiding a bug in portal - maybe portal changed the result format and shouldn't have, but that seemed like a low probability, but I should have asked - so mentioning it here.